### ISSUE TITLE
Allow setting Release via an MSBuild property

### ISSUE THREAD

**Issue by ericsampson** [getsentry/sentry-dotnet#3443]
### Problem Statement

Right now the Release value is constructed via other properties (if not set as an env var), but there are situations where it would be easier to allow the user to explicitly stamp the desired Release as a `SentryRelease` MSBuild prop.

**Comment by bitsandfoxes** [comment]
You can set the SentryRelease from code through the options. Does that not work for you?

**Comment by ericsampson** [comment]
> You can set the SentryRelease from code through the options. Does that not work for you?

No, because that doesn't work nicely in our CI pipeline. I'd like to be able to pass in the SentryRelease as an msbuild property argument to our `dotnet publish` command.

Does that make sense? Thanks!

**Comment by bitsandfoxes** [comment]
I see your point. You could set a custom release attribute to be available at runtime.
It would look a little something like this
```
<Target Name="EnsureCustomRelease" BeforeTargets="BeforeBuild">
  <MakeDir Directories="Properties" Condition="!Exists('Properties')" />
  <WriteLinesToFile File="Properties\CustomReleaseInfo.cs" Lines="using System.Reflection%3b" Overwrite="true" />
  <WriteLinesToFile File="Properties\CustomReleaseInfo.cs" Lines="[assembly: AssemblyMetadata(&quot;CustomRelease&quot;, &quot;$(CustomRelease)&quot;)]" Overwrite="false" />
</Target>
```
That can be retrieved like so
```
var assembly = Assembly.GetExecutingAssembly();
var customReleaseAttribute = assembly
    .GetCustomAttributes<AssemblyMetadataAttribute>()
    .FirstOrDefault(a => a.Key == "CustomRelease")
    ?.Value;
```
And you can set it via commandline `dotnet publish -p:CustomRelease=1.0.0`.
Would that unblock you?

**Comment by jamescrosswell** [comment]
This seems very similar to:
- of https://github.com/getsentry/sentry-dotnet/issues/2142

Is there anything we're suggesting in this issue that isn't already covered in the issue above?

**Comment by ericsampson** [comment]
@jamescrosswell I wasn't sure if the issue that you linked to included adding a new MSBuild property for `SentryRelease`.
If so, maybe we could update that issue's "spec" to call that out, and then this issue could be closed?

**Comment by bitsandfoxes** [comment]
Added this [here](https://github.com/getsentry/sentry-dotnet/issues/2142#issuecomment-2201125539) to keep track of it. Thanks @ericsampson for all the valuable input! It's much appreciated.

**Comment by ericsampson** [comment]
thank you @bitsandfoxes <3
